atlasdb-jepsen-tests/src/test/java/com/palantir/atlasdb/jepsen/lock/LockCorrectnessCheckerTest.java

```diff
@@ -0,0 +1,150 @@
+                .add(TestEventUtils.lockSuccess(3, process2))
+                .add(TestEventUtils.invokeUnlock(4, process1))
+                .add(TestEventUtils.unlockSuccess(5, process1))
+                .build();
+        CheckerResult result = runLockCorrectnessChecker(eventList);
+        assertThat(result.valid()).isFalse();
+        assertThat(result.errors()).containsExactly(eventList.get(2), eventList.get(3));
+    }
+
+    @Test
+    public void failedLockIntervalCoveredByAnotherProcessSucceeds() {
+        ImmutableList<Event> eventList = ImmutableList.<Event>builder()
+                .add(TestEventUtils.invokeLock(0, process1))
+                .add(TestEventUtils.lockSuccess(1, process1))
+                .add(TestEventUtils.invokeLock(2, process2))
+                .add(TestEventUtils.lockFailure(3, process2))
+                .add(TestEventUtils.invokeRefresh(4, process1))
+                .add(TestEventUtils.refreshSuccess(5, process1))
+                .build();
+        CheckerResult result = runLockCorrectnessChecker(eventList);
+        assertThat(result.valid()).isTrue();
+        assertThat(result.errors()).isEmpty();
+    }
+
+    @Test
+    public void shouldSucceedWhenThereIsASmallWindow() {
+        ImmutableList<Event> eventList = ImmutableList.<Event>builder()
+                .add(TestEventUtils.invokeLock(0, process1))
+                .add(TestEventUtils.lockSuccess(1, process1))
+                .add(TestEventUtils.invokeLock(2, process2))
+                .add(TestEventUtils.invokeLock(2, process3))
+                .add(TestEventUtils.invokeRefresh(3, process1))
+                .add(TestEventUtils.refreshSuccess(3, process1))
+                .add(TestEventUtils.lockSuccess(3, process3))
+                .add(TestEventUtils.lockSuccess(4, process2))
+                .add(TestEventUtils.invokeRefresh(5, process3))
+                .add(TestEventUtils.refreshSuccess(6, process3))
+                .build();
+        CheckerResult result = runLockCorrectnessChecker(eventList);
+        assertThat(result.valid()).isTrue();
+        assertThat(result.errors()).isEmpty();
+    }
+
+    @Test
+    public void shouldSucceedForIntervalEdgeCases() {
+        ImmutableList<Event> eventList = ImmutableList.<Event>builder()
+                .add(TestEventUtils.invokeLock(0, process1))
+                .add(TestEventUtils.lockSuccess(1, process1))
+                .add(TestEventUtils.invokeLock(1, process2))
+                .add(TestEventUtils.lockSuccess(2, process2))
+                .add(TestEventUtils.invokeLock(3, process3))
+                .add(TestEventUtils.lockSuccess(4, process3))
+                .add(TestEventUtils.invokeRefresh(4, process1))
+                .add(TestEventUtils.refreshSuccess(5, process1))
+                .build();
+        CheckerResult result = runLockCorrectnessChecker(eventList);
+        assertThat(result.valid()).isTrue();
+        assertThat(result.errors()).isEmpty();
```

Comment:
This is a good test, but we should hopefully give a brief comment or two explaining _why_ the expected behaviour is that this is correct (i.e. you can have 2 get the lock, drop it, 1 get the lock, and 1 refresh just before it's lost and 3 gets the lock after that).